build/pom.xml

```diff
@@ -23,4 +23,33 @@
     <module>org.eclipse.pde.build</module>
     <module>org.eclipse.pde.build.tests</module>
   </modules>
+  
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.eclipse.tycho</groupId>
+        <artifactId>tycho-surefire-plugin</artifactId>
+        <configuration>
+          <dependencies>
+            <dependency>
+              <type>eclipse-plugin</type>
+              <artifactId>org.eclipse.osgi.compatibility.state</artifactId>
+              <version>0.0.0</version>
+            </dependency>
+            <dependency>
+              <type>p2-installable-unit</type>
+              <artifactId>org.eclipse.equinox.executable</artifactId>
+              <version>0.0.0</version>
+            </dependency>
+            <dependency>
+              <type>eclipse-feature</type>
+              <artifactId>org.eclipse.sdk</artifactId>
+              <version>0.0.0</version>
+            </dependency>
+          </dependencies>
+          <testRuntime>p2Installed</testRuntime>
+        </configuration>
```

Comment:
Is this really necessary to have it installed at that stage (my concern is that test execution in requirements aren't so portable)?
Could it be replaced by some instructions in p2.inf?
Could it be replaced by creation of a TargetPlatform in the tests that would consist of running target?
Do you really need it as "p2installed", or would running this setting the right `testProduct/testApplication` to usual SDK values be sufficient? `p2installed` is usually useful only when there are crazy things like touchpoints or when you need the files to be organized on disk similarly to an installation.

Reply:
> Is this really necessary to have it installed at that stage (my concern is that test execution in requirements aren't so portable)?

I tried without but didn't work so well... I have no clue why the `compatibility` is not discovered automatically but it seems a common problem.

> Could it be replaced by some instructions in p2.inf?

Probably yes, but last time I tried it wasn't that obvious and introduced build cycles in aggregator.

> Could it be replaced by creation of a TargetPlatform in the tests that would consist of running target?

The problem is that the test don't start at all, the only good think is that one can easily try this standalone and play around by go into the folder `eclipse.pde/build/org.eclipse.pde.build.tests` and execute `mvn clean verify` and it is also quite fast.

> Do you really need it as "p2installed"

`p2installed` was just something I tried here because the test do special stuff, especially require that the plugins are exploded, that an installation location is set and so on, I haven't debugged this in detail and there might be mitigations but as pde-build is nothing I have experience with its hard for me to decide.

> p2installed is usually useful only when there are crazy things like touchpoints or when you need the files to be organized on disk similarly to an installation.

I fear that's the case here...

Reply:
> I have no clue why the compatibility is not discovered automatically but it seems a common problem.

It's a fragment of org.eclipse.osgi, and it's required whenever one invokes `Platform.getPlatformAdmin()`.

I added it (in p2.inf) and see that it indeed seems to require a full installation (using the simpleconfigurator/bundles.info file).

But I see the text.xml file contains 2 launches, one with a system property. Passing this system property may help running the tests here.
I'm actually curious as to whether PDE would handle this case as well without enabling p2.